File: server/src/propertie/dto/search-property.dto.ts
import { ApiPropertyOptional } from '@nestjs/swagger';

export class SearchPropertyDto {
  @ApiPropertyOptional({ description: 'Місто', example: 'Marbella' })
  city?: string;

  @ApiPropertyOptional({ description: 'Тип оголошення', enum: ['sale', 'rent'], example: 'sale' })
  listingType?: 'sale' | 'rent';

  @ApiPropertyOptional({ description: 'Тип нерухомості', example: 'villa' })
  propertyType?: string;

  @ApiPropertyOptional({ description: 'Мінімальна ціна', example: 200000 })
  priceFrom?: number;

  @ApiPropertyOptional({ description: 'Максимальна ціна', example: 800000 })
  priceTo?: number;

  @ApiPropertyOptional({ description: 'Номер сторінки', example: 1 })
  page?: number;

  @ApiPropertyOptional({ description: 'Кількість записів на сторінку', example: 10 })
  limit?: number;
}

File: real_estate_nikas_mobile/lib/data/models/search_property_dto.dart
class SearchPropertyDto {
  final String? city;
  final String? listingType;
  final String? propertyType;
  final int? priceFrom;
  final int? priceTo;
  final int page;
  final int limit;

  SearchPropertyDto({
    this.city,
    this.listingType,
    this.propertyType,
    this.priceFrom,
    this.priceTo,
    this.page = 1,
    this.limit = 10,
  });

  Map<String, String> toQueryParameters() {
    final params = <String, String>{};
    if (city != null) params['city'] = city!;
    if (listingType != null) params['listingType'] = listingType!;
    if (propertyType != null) params['propertyType'] = propertyType!;
    if (priceFrom != null) params['priceFrom'] = priceFrom.toString();
    if (priceTo != null) params['priceTo'] = priceTo.toString();
    params['page'] = page.toString();
    params['limit'] = limit.toString();
    return params;
  }
}
